crates/miden-lib/src/transaction/mod.rs

```diff
@@ -280,31 +294,56 @@ impl TransactionKernel {
     ///
     /// The output stack is expected to be arrange as follows:
     ///
-    /// Stack: [OUTPUT_NOTES_COMMITMENT, FINAL_ACCOUNT_COMMITMENT, tx_expiration_block_num]
+    /// Stack: [OUTPUT_NOTES_COMMITMENT, ACCOUNT_UPDATE_COMMITMENT, tx_expiration_block_num]
     ///
     /// Where:
     /// - OUTPUT_NOTES_COMMITMENT is the commitment of the output notes.
-    /// - FINAL_ACCOUNT_COMMITMENT is the final account commitment of the account that the
-    ///   transaction is being executed against.
+    /// - ACCOUNT_UPDATE_COMMITMENT is the hash of the final account commitment and the account
+    ///   delta commitment of the account that the transaction is being executed against.
     /// - tx_expiration_block_num is the block height at which the transaction will become expired,
     ///   defined by the sum of the execution block ref and the transaction's block expiration delta
     ///   (if set during transaction execution).
     ///
     /// The actual data describing the new account state and output notes is expected to be located
     /// in the provided advice map under keys `OUTPUT_NOTES_COMMITMENT` and
+    /// `ACCOUNT_UPDATE_COMMITMENT`, where the final data for the account state is located under
     /// `FINAL_ACCOUNT_COMMITMENT`.
     pub fn from_transaction_parts(
         stack: &StackOutputs,
         adv_map: &AdviceMap,
         output_notes: Vec<OutputNote>,
     ) -> Result<TransactionOutputs, TransactionOutputError> {
-        let (final_account_commitment, output_notes_commitment, expiration_block_num) =
+        let (account_update_commitment, output_notes_commitment, expiration_block_num) =
             Self::parse_output_stack(stack)?;
 
+        let account_update_data = adv_map.get(&account_update_commitment).ok_or_else(|| {
+            TransactionOutputError::AccountUpdateCommitment(
+                "failed to find ACCOUNT_UPDATE_COMMITMENT in advice map",
+            )
+        })?;
+
+        if account_update_data.len() != 8 {
+            return Err(TransactionOutputError::AccountUpdateCommitment(
+                "expected account update commitment advice map entry to contain exactly 8 elements",
+            ));
+        }
+
+        // SAFETY: We just asserted that the data is of length 8 so slicing the data into two words
+        // is fine.
+        let final_account_commitment = Digest::from(
+            <[Felt; 4]>::try_from(&account_update_data[0..4])
+                .expect("we should have sliced off exactly four elements"),
+        );
+        let account_delta_commitment = Digest::from(
+            <[Felt; 4]>::try_from(&account_update_data[4..8])
+                .expect("we should have sliced off exactly four elements"),
+        );
```

Comment:
Should there be a sanity check here that hashing these results in the same value that was on the stack or would it be redundant?